src/compiler/checker.ts

```diff
@@ -8768,8 +8768,15 @@ namespace ts {
             }
             if (symbol.flags & SymbolFlags.ModuleExports) {
                 const fileSymbol = getSymbolOfNode(getSourceFileOfNode(symbol.valueDeclaration));
+                const result = createSymbol(fileSymbol.flags, "exports" as __String);
+                result.declarations = fileSymbol.declarations ? fileSymbol.declarations.slice() : [];
+                result.parent = symbol;
+                result.target = fileSymbol;
+                if (fileSymbol.valueDeclaration) result.valueDeclaration = fileSymbol.valueDeclaration;
+                if (fileSymbol.members) result.members = new Map(fileSymbol.members);
+                if (fileSymbol.exports) result.exports = new Map(fileSymbol.exports);

```

Comment:
How much of a memory hit for big checkJs projects (that use `module.exports`) is this?

Reply:
The only baseline we have for this is "does chrome-devtools-frontend run out of memory?". I'll run the user tests.

A priori, it shouldn't be gigantic; a file with 1000 exports will add 1000 map entries. The first page of google only returns [an SO answer from a V8 developer](https://stackoverflow.com/a/54466812/7851), which points out that each entry has key,value and next pointers. So...is that 6000 bytes per large file? It's not too expensive, but on the other hand this is a mostly cosmetic feature. The exception is the bug that the PR fixes.